### PR TITLE
Stabilize UdpSocket::peer_addr

### LINKED ISSUE #59127: Tracking issue for udp_peer_addr
Tracking issue for making `peer_addr()` on `UdpSocket` stable: #59106, #59104

### PR DESCRIPTION
**Pull request by messense**
Fixes #59127

### CODE REVIEW

File: src/libstd/net/udp.rs [hunk 1]
```diff
@@ -199,14 +198,13 @@ impl UdpSocket {
...
     /// ```no_run
-    /// #![feature(udp_peer_addr)]
     /// use std::net::UdpSocket;
     ///
     /// let socket = UdpSocket::bind("127.0.0.1:34254").expect("couldn't bind to address");
     /// assert_eq!(socket.peer_addr().unwrap_err().kind(),
     ///            ::std::io::ErrorKind::NotConnected);
     /// ```
-    #[unstable(feature = "udp_peer_addr", issue = "59127")]
+    #[stable(feature = "udp_peer_addr", since = "1.39.0")]
```
Comment (hellow554) [inline src/libstd/net/udp.rs]: @jonas-schievink tagged this as 1.40, but this is saying 1.39

Comment (Centril) [reply src/libstd/net/udp.rs]: You need to account for +10 days from beta promotion due to FCP. Usually 15 days to be on the safe side as 10 is awfully close.

Comment (Centril) [inline src/libstd/net/udp.rs]:
```suggestion
    #[stable(feature = "udp_peer_addr", since = "1.40.0")]
```

**Comment by Centril** [comment]
Picking randomly from T-libs... r? @dtolnay

**Review by dtolnay — APPROVED** [review]
Thanks!

**Comment by dtolnay** [comment]
Libs fcp was already completed in https://github.com/rust-lang/rust/issues/59127#issuecomment-524586657.

@bors r+

**Pull request merged**